File: src/Model/Session/GeneralCommandType.java
package MediaBrowser.Model.Session;

/** 
 This exists simply to identify a set of known commands.
*/
public enum GeneralCommandType
{
	MoveUp(0),
	MoveDown(1),
	MoveLeft(2),
	MoveRight(3),
	PageUp(4),
	PageDown(5),
	PreviousLetter(6),
	NextLetter(7),
	ToggleOsd(8),
	ToggleContextMenu(9),
	Select(10),
	Back(11),
	TakeScreenshot(12),
	SendKey(13),
	SendString(14),
	GoHome(15),
	GoToSettings(16),
	VolumeUp(17),
	VolumeDown(18),
	Mute(19),
	Unmute(20),
	ToggleMute(21),
	SetVolume(22),
	SetAudioStreamIndex(23),
	SetSubtitleStreamIndex(24),
	ToggleFullscreen(25),
	DisplayContent(26),
	GoToSearch(27),
	DisplayMessage(28);

	private int intValue;
	private static java.util.HashMap<Integer, GeneralCommandType> mappings;
	private static java.util.HashMap<Integer, GeneralCommandType> getMappings()
	{
		if (mappings == null)
		{
			synchronized (GeneralCommandType.class)
			{
				if (mappings == null)
				{
					mappings = new java.util.HashMap<Integer, GeneralCommandType>();
				}
			}
		}
		return mappings;
	}

	private GeneralCommandType(int value)
	{
		intValue = value;
		getMappings().put(value, this);
	}

	public int getValue()
	{
		return intValue;
	}

	public static GeneralCommandType forValue(int value)
	{
		return getMappings().get(value);
	}
}

File: jellyfin-model/src/commonMain/kotlin/org/jellyfin/sdk/model/socket/GeneralCommandType.kt
package org.jellyfin.sdk.model.socket

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
public enum class GeneralCommandType {
	@SerialName("MoveUp")
	MoveUp,

	@SerialName("MoveDown")
	MoveDown,

	@SerialName("MoveLeft")
	MoveLeft,

	@SerialName("MoveRight")
	MoveRight,

	@SerialName("PageUp")
	PageUp,

	@SerialName("PageDown")
	PageDown,

	@SerialName("PreviousLetter")
	PreviousLetter,

	@SerialName("NextLetter")
	NextLetter,

	@SerialName("ToggleOsd")
	ToggleOsd,

	@SerialName("ToggleContextMenu")
	ToggleContextMenu,

	@SerialName("Select")
	Select,

	@SerialName("Back")
	Back,

	@SerialName("TakeScreenshot")
	TakeScreenshot,

	@SerialName("SendKey")
	SendKey,

	@SerialName("SendString")
	SendString,

	@SerialName("GoHome")
	GoHome,

	@SerialName("GoToSettings")
	GoToSettings,

	@SerialName("VolumeUp")
	VolumeUp,

	@SerialName("VolumeDown")
	VolumeDown,

	@SerialName("Mute")
	Mute,

	@SerialName("Unmute")
	Unmute,

	@SerialName("ToggleMute")
	ToggleMute,

	@SerialName("SetVolume")
	SetVolume,

	@SerialName("SetAudioStreamIndex")
	SetAudioStreamIndex,

	@SerialName("SetSubtitleStreamIndex")
	SetSubtitleStreamIndex,

	@SerialName("ToggleFullscreen")
	ToggleFullscreen,

	@SerialName("DisplayContent")
	DisplayContent,

	@SerialName("GoToSearch")
	GoToSearch,

	@SerialName("DisplayMessage")
	DisplayMessage,

	@SerialName("SetRepeatMode")
	SetRepeatMode,

	@SerialName("ChannelUp")
	ChannelUp,

	@SerialName("ChannelDown")
	ChannelDown,

	@SerialName("SetMaxStreamingBitrate")
	SetMaxStreamingBitrate,

	@SerialName("Guide")
	Guide,

	@SerialName("ToggleStats")
	ToggleStats,

	@SerialName("PlayMediaSource")
	PlayMediaSource,

	@SerialName("PlayTrailers")
	PlayTrailers,
}
